src/main/java/io/nats/client/impl/NatsDispatcher.java

```diff
@@ -207,8 +285,28 @@ public Dispatcher unsubscribe(String subject, int after) {
         if (subject == null || subject.length() == 0) {
             throw new IllegalArgumentException("Subject is required in unsubscribe");
         }
-        
-        NatsSubscription sub = subscriptions.get(subject);
+
+        NatsSubscription sub = this.subscriptionsUsingDefaultHandler.get(subject);
+
+        if (sub != null) {
+            this.connection.unsubscribe(sub, after); // Connection will tell us when to remove from the map
+        }
+
+        return this;
+    }
+
+    public Dispatcher unsubscribe(Subscription subscription, int after) {
+        if (subscription.getDispatcher() != this) {
```

Comment:
last thing - this needs
```
if (!this.running.get()) {
            throw new IllegalStateException("Dispatcher is closed");
        }

        if (isDraining()) { // No op while draining
            return this;
        }
```
at the top like the previous method

Reply:
Done! Good find!